examples/dagger/test/coffee/BasicTest.kt

```diff
@@ -0,0 +1,21 @@
+package coffee
+
+import com.google.common.truth.Truth.assertThat
+import heating.ElectricHeater
+import org.junit.Test
+import time.Delayer
+
+class BasicTest {
+    @Test fun FooTest() {
+        val heater = ElectricHeater(object : Delayer {
+            override fun delay() {
```

Comment:
as i much as i like the dagger heater example i think https://github.com/facebook/buck/pull/2197/files#diff-8dbee0235ce31d15e79389735d429850 made it easier to understand..

Reply:
Oh yeah, that Buck's example/test seems great - easy to follow (seems like couldn't be any easier).

Reply:
Possibly, however the example isn't just an example, it's also a test. The buck example doesn't actually adequately exercise the use cases.

Reply:
Though, I'll rework the names a bit for clarity.